### PR TITLE
fix: [CDS-69297]: doing fixes for customer cases

### PR DESCRIPTION
**Pull request by TarunUba1**
## Describe your changes
using shell script code to run the commands to enable mult-line scripts and also ability to print env variables.
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/47793)

### CODE REVIEW

**Comment by TarunUba1** [comment]
trigger uts

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/helm/HelmDeployServiceImplNG.java [hunk 1]
```diff
@@ -215,8 +216,9 @@ public HelmCommandResponseNG deploy(HelmInstallCommandRequestNG commandRequest)
           ServiceHookType.PRE_HOOK, ServiceHookAction.FETCH_FILES, serviceHookDTO.getWorkingDirectory(), logCallback);
       kubernetesConfig = containerDeploymentDelegateBaseHelper.createKubernetesConfig(
           commandRequest.getK8sInfraDelegateConfig(), commandRequest.getWorkingDir(), logCallback);
-
-      prepareRepoAndCharts(commandRequest, commandRequest.getTimeoutInMillis(), logCallback);
+      String helmRepoConfigPath =
+          prepareRepoAndCharts(commandRequest, commandRequest.getTimeoutInMillis(), logCallback);
+      serviceHookHandler.addToContext(HELM_REPOSITORY_CONFIG_PATH.getContextName(), helmRepoConfigPath);
```
Comment (abosii-harness) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/helm/HelmDeployServiceImplNG.java:221]: As discussed, due to that repo cache folder is cleaned up after fetch it doesn't really make sense to add it to context

File: 960-api-services/src/main/java/io/harness/shell/ScriptProcessExecutor.java [hunk 2]
```diff
@@ -345,8 +350,10 @@ protected void processLine(String line) {
       executionDataBuilder.sweepingOutputEnvVariables(envVariablesMap);
 
       commandExecutionStatus = processResult.getExitValue() == 0 ? SUCCESS : FAILURE;
-      if (commandExecutionStatus == SUCCESS) {
+      if (commandExecutionStatus == SUCCESS && denoteOverallSuccess) {
```
Comment (abosii-harness) [inline 960-api-services/src/main/java/io/harness/shell/ScriptProcessExecutor.java:353]: Let's add a TODO comment to figure out what's going on here. This logic is fine for now, but there should've been used closeLogStream parameter in shell script config. Something is broken here

**Review by abosii-harness — CHANGES_REQUESTED** [review]

File: 960-api-services/src/main/java/io/harness/shell/ScriptProcessExecutor.java (hunk 2 above)
Comment (TarunUba1) [reply 960-api-services/src/main/java/io/harness/shell/ScriptProcessExecutor.java:354]: Done

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/helm/HelmDeployServiceImplNG.java (hunk 1 above)
Comment (TarunUba1) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/task/helm/HelmDeployServiceImplNG.java]: removed

**Comment by TarunUba1** [comment]
trigger uts

**Review by namantalaycha10 — APPROVED** [review]

**Review by GoelPratyush99 — APPROVED** [review]

**Comment by TarunUba1** [comment]
trigger review

**Pull request merged**